flocker/acceptance/endtoend/test_dataset.py

```diff
@@ -35,6 +40,112 @@ def test_dataset_creation(self, cluster):
         """
         return create_dataset(self, cluster)
 
+    def _get_package_source(self, default_version=None):
+        """
+        Get the package source for the flocker version under test from
+        environment variables.
+
+        :param unicode default_version: The version of flocker to use
+            if not specified in environment variables.
+
+        :return: A ``PackageSource`` that can be used to install the version of
+            flocker under test.
+        """
+        env_vars = ['FLOCKER_ACCEPTANCE_PACKAGE_BRANCH',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_VERSION',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_BUILD_SERVER']
+        missing_vars = list(x for x in env_vars if x not in os.environ)
+        if missing_vars:
+            raise SkipTest(
+                'Missing environment variables for upgrade test: %s.' %
+                ', '.join(missing_vars))
+        version = (os.environ['FLOCKER_ACCEPTANCE_PACKAGE_VERSION'] or
+                   default_version)
```

Comment:
Hrm so I've taken the stance that all of these env vars must be defined in order for the test not to be skipped, and that `None` values from https://github.com/ClusterHQ/flocker/blob/741f880d4171a892c6d559270bb00ce781833150/flocker/provision/_common.py#L16 get turned into empty strings. (see https://github.com/ClusterHQ/flocker/pull/2547/files#diff-d7d14368f6d81514989386a295faa983R120).

Some of this was motivated by https://github.com/ClusterHQ/flocker/blob/741f880d4171a892c6d559270bb00ce781833150/flocker/provision/_install.py#L324 which suggests that empty strings have the same semantics as `None`. So I'd have to write more intelligent code when setting the environment variables to remove them if they are empty strings.

The other part was trying to protect against typos if people were specifying the environment variable by hand. It would be sort of lousy for:

```
export FLOCKER_ACCEPTANCE_PACKAGE_VERSOIN=1.8.0-1-322
```

to result in testing against the default version, when that's probably not the intent. Instead I'd rather error out letting me know that I forgot to specify `FLOCKER_ACCEPTANCE_PACKAGE_VERSION`.

So I'd sort of like to keep those semantics (all three must be set, and empty strings mean use the defaults), but I'll add additional information to the docstrings and enhance the error reporting to make that more obvious.